packages/dara-core/js/shared/interactivity/data-variable.tsx

```diff
@@ -286,8 +286,8 @@ export function useFetchDerivedDataVariable(
             // For derived data variables count can only be fetched when task is not running so we have to make the request here
             // As the total count could have changed because of the underlying DV changing
             const [totalCount, schema = null] = await Promise.all([
-                fetchDataVariableCount(variable.uid, extras, mergedFilters),
-                ...(withSchema ? [fetchDataVariableSchema(variable.uid, extras)] : []),
+                fetchDataVariableCount(variable.uid, extras, mergedFilters, dvValue.cache_key),
+                ...(withSchema ? [fetchDataVariableSchema(variable.uid, extras, dvValue.cache_key)] : []),
             ] as const);
```

Comment:
Do we not have any tests that ensure that at least we make the right request to the backend with the data? Might help prevent mistakes here in the future

Reply:
Good point. We had a test, but it was not checking for the `cache_key` presense. Overrode the handler for DDV that specifically checks for the key there.

Also, some types were missing the schema options flag, fixed those as well.